raiden/ui/web/src/app/components/token-network-selector/token-network-selector.component.ts

```diff
@@ -0,0 +1,149 @@
+import { Component, EventEmitter, forwardRef, OnInit, Output } from '@angular/core';
+import {
+    AbstractControl,
+    ControlValueAccessor,
+    FormControl, NG_VALIDATORS,
+    NG_VALUE_ACCESSOR,
+    ValidationErrors,
+    Validator,
+    ValidatorFn
+} from '@angular/forms';
+import { from, Observable } from 'rxjs';
+import { filter, flatMap, map, share, startWith, toArray } from 'rxjs/operators';
+import { UserToken } from '../../models/usertoken';
+import { RaidenService } from '../../services/raiden.service';
+import { amountToDecimal } from '../../utils/amount.converter';
+
+@Component({
+    selector: 'app-token-network-selector',
+    templateUrl: './token-network-selector.component.html',
+    styleUrls: ['./token-network-selector.component.css'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => TokenNetworkSelectorComponent),
+            multi: true
+        },
+        {
+            provide: NG_VALIDATORS,
+            useExisting: forwardRef(() => TokenNetworkSelectorComponent),
+            multi: true
+        }
+    ]
+})
+export class TokenNetworkSelectorComponent implements OnInit, ControlValueAccessor, Validator {
+
+    @Output() valueChanged = new EventEmitter<UserToken>();
+
+    private tokens$: Observable<UserToken[]>;
+    public filteredOptions$: Observable<UserToken[]>;
+
+    readonly tokenFc = new FormControl('', [this.addressValidatorFn(this.raidenService)]);
+
+    constructor(private raidenService: RaidenService) {
+    }
+
+    ngOnInit() {
+        this.tokens$ = this.raidenService.getTokens(true).pipe(
+            flatMap((tokens: UserToken[]) => from(tokens)),
+            toArray(),
+            map(value => value.sort(this._compareTokens)),
+            share()
+        );
+
+        this.filteredOptions$ = this.tokenFc.valueChanges.pipe(
+            startWith(''),
+            flatMap(value => this._filter(value))
+        );
+    }
+
+    // noinspection JSMethodCanBeStatic
+    trackByFn(token: UserToken): string {
+        return token.address;
+    }
+
+    tokenSelected(value: UserToken) {
+        this.tokenFc.setValue(value.address);
+    }
+
+    // noinspection JSMethodCanBeStatic
+    private _compareTokens(a: UserToken, b: UserToken): number {
+        if (a.connected === b.connected) {
```

Comment:
Take care with objects comparison: if you don't ensure they're actually the same object reference, you'll get false here (differently from python' dicts, js objects aren't compared by member's values). If it's false, you may get unexpected behavior on that `a.connected ?` comparison below.

Reply:
Thanks. For some reason I was treating them as booleans  I will change it